### ISSUE TITLE
Error in initializing the CognitoCachingCredentialsProvider introduced in 2.12.3

### ISSUE THREAD

**Issue by agalfioni** [aws-amplify/aws-sdk-android#780]
**Describe the bug**
Since 2.12.3 version, i can't initialize CognitoCachingCredentialsProvider using constructor `public CognitoCachingCredentialsProvider(Context context, String identityPoolId, Regions region)`

I obtain the next error from catch block in initialize(Context context) method:

> E/CognitoCachingCredentialsProvider: Error in initializing the CognitoCachingCredentialsProvider. java.lang.ClassCastException: java.lang.Long cannot be cast to java.lang.String

And later:

> java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.amazonaws.internal.keyvaluestore.AWSKeyValueStore.get(java.lang.String)' on a null object reference
> at com.amazonaws.auth.CognitoCachingCredentialsProvider.getCachedIdentityId(CognitoCachingCredentialsProvider.java:574)
> 
> 

when execute `credentialsProvider.getCachedIdentityId()` with previously instantiate CognitoCachingCredentialsProvider.

**To Reproduce**
Just try to instantiate CognitoCachingCredentialsProvider.
```
String COGNITO_POOL_ID = "us-east-2:_________"; // unauthenticated pool with AWS IoTManager permissions.
Regions MY_REGION = Regions.US_EAST_2;

new CognitoCachingCredentialsProvider(
                getApplicationContext(),
                COGNITO_POOL_ID,
                MY_REGION
        );
```

**Which AWS service(s) are affected?**

cognitoidentityprovider

**Expected behavior**
I expect to be able to instantiate CognitoCachingCredentialsProvider.

**Screenshots**

**Environment Information (please complete the following information):**
 - AWS Android SDK Version: 2.12.3
 - Device: Real device, Xiaomi Redmi 5 Plus
 - Android Version: 8.1.0 API 27
 - Specific to simulators: No

**Comment by kvasukib** [comment]
@agalfioni Sorry for the inconvenience caused. We are looking into reproducing and fixing the issue.

**Comment by kvasukib** [comment]
@agalfioni I have submitted a PR #781 to address this issue. Please feel free to build from the PR to test and see if it avoids the crash.

**Comment by kvasukib** [comment]
@agalfioni The fix has been released in `2.12.4` version of the SDK. Please refer to [CHANGELOG](https://github.com/aws-amplify/aws-sdk-android/blob/master/CHANGELOG.md#release-2124) for more information.

**Comment by agalfioni** [comment]
Thanks @kvasukib the bug is fixed with the newer version of the SDK.

Regards!
